bulk_indexer.go

```diff
@@ -336,12 +336,7 @@ func (b *bulkIndexer) Flush(ctx context.Context) (BulkIndexerResponseStat, error
 				b.retryCounts[b.itemsAdded] = count
 
 				if b.gzipw != nil {
-					// First loop, read from the gzip reader
-					if len(buf) == 0 {
-						n, _ := gr.Read(buf[:cap(buf)])
-						buf = buf[:n]
-					}
-
+					// newlines in the current buf
 					newlines := bytes.Count(buf, []byte{'\n'})
 
 					// loop until we've seen the start newline
```

Comment:
Now the for loop below ("loop until we've seen the start newline") isn't executed if the first document needs to be read (because `startln` == 0), which means the logic isn't very straightforward as it deviates from what the comments say. The code might still work but it feels like an accident. That's the reason why I left the extra `len(buf) == 0` in the for loop in #129. It works without, but understanding why the code works without that is quite confusing.